packages/keypair/src/generate-vanity-key-pair.ts

```diff
@@ -0,0 +1,49 @@
+import { createKeyPairSignerFromPrivateKeyBytes, type KeyPairSigner } from '@solana/kit'
+
+export interface GenerateVanityKeyPairProps {
+  caseSensitive?: boolean
+  prefix?: string
+  suffix?: string
+}
+
+export async function generateVanityKeyPair({
+  caseSensitive = true,
+  prefix = '',
+  suffix = '',
+}: GenerateVanityKeyPairProps): Promise<KeyPairSigner> {
+  if (!prefix && !suffix) {
+    const privateKeyBytes = crypto.getRandomValues(new Uint8Array(32))
+    return createKeyPairSignerFromPrivateKeyBytes(privateKeyBytes, true)
+  }
+
+  // We can't report progress from here directly without a callback or generator
+  // But for the worker implementation, we will rewrite the loop there.
+  // This function stays as a utility for simple synchronous usage (though it's async due to createKeyPairSignerFromPrivateKeyBytes)
+
+  while (true) {
+    const privateKeyBytes = crypto.getRandomValues(new Uint8Array(32))
+    const signer = await createKeyPairSignerFromPrivateKeyBytes(privateKeyBytes, true)
+    const address = signer.address
+
+    let match = true
```

Comment:
I may be wrong here, but I feel this is easier to reason about if we start with `let match = false`, then make it true if we find a match.